File: others/WorksApplication/Worksap Package/workspace/China_Intern/China_Intern_Java/src/jp/co/worksap/intern/entities/product/ProductMstDtoReader.java
package jp.co.worksap.intern.entities.product;

import java.io.IOException;

import jp.co.worksap.intern.constants.Constants;
import jp.co.worksap.intern.constants.PriceUnitType;
import jp.co.worksap.intern.entities.AbstractDTOReader;

public class ProductMstDtoReader extends AbstractDTOReader<ProductMstDto> {
	private static final int COLUMN_INDEX_PRODUCT_ID = 0;
	private static final int COLUMN_INDEX_PRODUCT_TYPE = 1;
	private static final int COLUMN_INDEX_PRODUCT_NAME = 2;
	private static final int COLUMN_INDEX_PRICE = 3;
	private static final int COLUMN_INDEX_PRICE_UNIT = 4;

	private String fileAddress = Constants.DEFAULT_CSV_FOLDER
			+ "PRODUCT_MST.csv";

	/**
	 * use default file address
	 * 
	 * @throws IOException
	 */
	public ProductMstDtoReader() throws IOException {
		super(ProductMstDtoReader.class.getName());
		super.init();
	}

	/**
	 * use customize file address
	 * 
	 * @param fileAddress
	 * @throws IOException
	 */
	public ProductMstDtoReader(final String fileAddress) throws IOException {
		super(ProductMstDtoReader.class.getName());
		this.fileAddress = fileAddress;
		super.init();
	}

	@Override
	protected String getFileArress() {
		return fileAddress;
	}

	@Override
	protected ProductMstDto convertArrayToDTO(String[] value) throws IOException {

		Long productMstId = Long.valueOf(value[COLUMN_INDEX_PRODUCT_ID]);
		String productType = value[COLUMN_INDEX_PRODUCT_TYPE];
		String productName = value[COLUMN_INDEX_PRODUCT_NAME];
		double price = Double.parseDouble(value[COLUMN_INDEX_PRICE]);
		PriceUnitType priceUnit = PriceUnitType.valueOfString(value[COLUMN_INDEX_PRICE_UNIT]);

		ProductMstDto dto = new ProductMstDto(productMstId, productType, productName, price, priceUnit);
		return dto;
	}
}

File: others/WorksApplication/Worksap Package/workspace/China_Intern/China_Intern_Cpp/src/entities/product/product_mst_dto_reader.cpp
#include "product_mst_dto.h"
#include "product_mst_dto_reader.h"
#include "../../constants/constants.h"
#include "../icsv_master_dto.h"
#include <string>
#include <vector>
#include <cstdlib>
using namespace std;

ProductMstDTOReader::ProductMstDTOReader() {
	this->file_address = kDefaultCsvFolder + "PRODUCT_MST.csv";
	AbstractDTOReader::init();
}

ProductMstDTOReader::ProductMstDTOReader(string &file_address) {
	this->file_address = file_address;
	AbstractDTOReader::init();
}

string ProductMstDTOReader::get_file_address() {
	return this->file_address;
}

ICsvMasterDTO *ProductMstDTOReader::convert_array_to_dto(vector<string> &value) {
	long product_mst_id = atol(value[COLUMN_INDEX_PRODUCT_MST_ID].c_str());
	string product_type = value[COLUMN_INDEX_PRODUCT_TYPE];
	string product_name = value[COLUMN_INDEX_PRODUCT_NAME];
	double price = atof(value[COLUMN_INDEX_PRICE].c_str());
	PriceUnitType price_unit = price_unit_type_value_of_string(value[COLUMN_INDEX_PRICE_UNIT]);
	ICsvMasterDTO *obj = new ProductMstDTO(product_mst_id, product_type, product_name, price, price_unit);
	return obj;
}

ProductMstDTOReader::~ProductMstDTOReader() {

}

